ostd/src/mm/page_table/cursor.rs

```diff
@@ -734,26 +734,97 @@ where
         None
     }
 
-    pub fn preempt_guard(&self) -> &DisabledPreemptGuard {
-        &self.0.preempt_guard
-    }
-
-    /// Consumes itself and leak the root guard for the caller if it locked the root level.
+    /// Copies the mapping from the given cursor to the current cursor.
     ///
-    /// It is useful when the caller wants to keep the root guard while the cursor should be dropped.
-    pub(super) fn leak_root_guard(mut self) -> Option<PageTableNode<E, C>> {
-        if self.0.guard_level != C::NR_LEVELS {
-            return None;
-        }
+    /// All the mappings in the current cursor's range must be empty. The
```

Comment:
Does this requirement need to be placed under Safety?

Reply:
It's in the panic condition, under `# Panics`.